jit/jit-rules-x86-64.c

```diff
@@ -2749,6 +2730,84 @@ memory_copy(jit_gencode_t gen, unsigned char *inst,
 	return inst;
 }
 
+/*
+ * Set a small block. This code will be inlined.
+ * Set is_aligned to 0 if you don't know if the source and target locations
+ * are aligned on a 16byte boundary and != 0 if you know that both blocks are
+ * aligned.
+ * Set use_sse to 0 if you don't want to use SSE. Setting this to non-zero
+ * will make this function ignore src_xreg
+ * We assume that offset + size is in the range -2GB ... +2GB.
+ */
+static unsigned char *
+small_block_set(jit_gencode_t gen, unsigned char *inst,
+				int dreg, jit_nint doffset,
+				jit_nint val, jit_nint size,
+				int scratch_reg, int scratch_xreg,
+				int is_aligned, int use_sse)
+{
+	jit_nint offset = 0;
+
+	if(val == 0)
+	{
+		if(!use_sse || size % 16 != 0)
+		{
+			x86_64_clear_reg(inst, scratch_reg);
+		}
+	}
+	else
+	{
+		for(int i = 0; i < 8; i++)
+		{
+			val = (val << 8) | (val & 0xff);
```

Comment:
It probably doesn't matter that much in practice but `jit_nint` is a signed type and shifting into the sign bit is implementation-defined behavior.

Reply:
fixed in 2904804a1fb8e6986aa4f32521bcc6849d1a658c

Reply:
val &= 0xff;
val |= val << 8;
val |= val << 16;
val |= val << 32;